recidiviz/calculator/bq/bq_utils.py

```
@@ -0,0 +1,66 @@
+"""Helper functions for creating and updating BigQuery datasets/tables/views."""
+
+import logging
+
+from google.cloud import bigquery
+from google.cloud import exceptions
+
+# Importing only for typing.
+from recidiviz.calculator.bq import views
+
+
+_client = None
+def client() -> bigquery.Client:
+    global _client
+    if not _client:
+        _client = bigquery.Client()
+    return _client
+
+
+def create_dataset_if_necessary(dataset_ref: bigquery.dataset.DatasetReference):
+    """Create a BigQuery dataset if it does not exist."""
+    try:
+        client().get_dataset(dataset_ref)
+    except exceptions.NotFound:
+        logging.info(
+            'Dataset %s does not exist. Creating...', str(dataset_ref))
+        dataset = bigquery.Dataset(dataset_ref)
+        client().create_dataset(dataset)
```

Similar to my previous comment: could this raise another exception we should be catching or do we want to just let these classes of potential IO exceptions just bubble?

---

I think it's fine to let them bubble! Not sure what other exceptions may be raised, but I think it's fine to wait and find out. I can catch all exceptions if you'd prefer, but if there's other unexpected behavior I think it should fail loudly here.